internal/engine/buildcontroller.go

```diff
@@ -143,15 +149,21 @@ func (c *BuildController) buildAndDeploy(ctx context.Context, st store.RStore, e
 	return c.b.BuildAndDeploy(ctx, st, targets, entry.buildStateSet)
 }
 
-// cancel any in-progress builds associated with disabled UIResources
+// cancel any in-progress builds associated with canceled builds and disabled UIResources
 // when builds are fully represented by api objects, cancellation should probably
 // be tied to those rather than the UIResource
-func (c *BuildController) cleanupDisabledBuilds(st store.RStore) {
+func (c *BuildController) cleanUpCanceledBuilds(st store.RStore) {
 	state := st.RLockState()
 	defer st.RUnlockState()
 
 	for _, ms := range state.ManifestStates() {
-		if ms.DisableState == v1alpha1.DisableStateDisabled {
+		disabled := ms.DisableState == v1alpha1.DisableStateDisabled
+		canceled := false
+		if cancelButton, ok := state.UIButtons[uibutton.CancelButtonName(ms.Name.String())]; ok {
+			lastCancelClick := cancelButton.Status.LastClickedAt
+			canceled = !lastCancelClick.Time.Before(ms.CurrentBuild.StartTime)
```

Comment:
Also not critical given how it's used here - just sharing in case you haven't seen before, the `timecmp` package can be useful when dealing with the API times, particularly if comparing to stdlib times (see package doc for long-winded explanation)

https://github.com/tilt-dev/tilt/blob/449fe2dbcc77c7bfacea82d1b0e14f93c4b0194a/internal/timecmp/cmp.go#L58-L66

Reply:
oh, nice